lib/matplotlib/tests/test_polar.py

```diff
@@ -436,6 +436,28 @@ def test_cursor_precision():
     assert ax.format_coord(2, 1) == "θ=0.637π (114.6°), r=1.000"
 
 
+def test_custom_fmt_ydata():
+    ax = plt.subplot(projection="polar")
+    def millions(x):
+        return '$%1.1fM' % (x*1e-6)
+
+    ax.fmt_ydata = millions
+    assert ax.format_coord(12, 2e7) == "θ=3.8197186342π (687.54935416°), r=$20.0M"
+    assert ax.format_coord(1234, 2e6) == "θ=392.794399551π (70702.9919191°), r=$2.0M"
+    assert ax.format_coord(3, 100) == "θ=0.95493π (171.887°), r=$0.0M"
+
+
+def test_custom_fmt_xdata():
+    ax = plt.subplot(projection="polar")
+    def millions(x):
+        return '$%1.1fM' % (x*1e-6)
+
+    ax.fmt_xdata = millions
+    assert ax.format_coord(2e5, 1) == "θ=$0.2M, r=1.000"
+    assert ax.format_coord(1, .1) == "θ=$0.0M, r=0.100"
+    assert ax.format_coord(1e6, 0.005) == "θ=$1.0M, r=0.005"
```

Comment:
Could you also add a test for the case with both fmt_xdata and fmt_ydata set?

It might be easier to do this all within a single test and add comments for the various cases.

```suggestion
def test_custom_fmt_xdata():
    ax = plt.subplot(projection="polar")
    def millions(x):
        return '$%1.1fM' % (x*1e-6)


    # Test only x formatter
    ax.fmt_xdata = millions
    ax.fmt_ydata = None
    assert ax.format_coord(2e5, 1) == "θ=$0.2M, r=1.000"
    assert ax.format_coord(1, .1) == "θ=$0.0M, r=0.100"
    assert ax.format_coord(1e6, 0.005) == "θ=$1.0M, r=0.005"
    
    # Test only y formatter
    ax.fmt_xdata = None
    ax.fmt_ydata = millions
    ...
    
    # Test both x and y formatters
    ax.fmt_xdata = millions
    ax.fmt_ydata = millions
    ...
```